### ISSUE TITLE
[Fix/Internal] [forecasting] data is shuffled when splitting the data to train/test

### ISSUE THREAD

**Issue by lisphilar** [lisphilar/covid19-sir#802]
## Summary
In `_RegressorBase` (internal class for forecasting), the dataset regarding parmeter values and delayed indicators is shuffled when splitting it to training and test data. This is related to #795.

## Codes

```Python
import covsirphy as cs
# Dataset preparation
data_loader = cs.DataLoader("input")
jhu_data = data_loader.jhu()
# Scenario analysis
snl = cs.Scenario(country="Country name used")
snl.register(jhu_data)
snl.trend()
snl.estimate(cs.SIRF)
snl.fit()
```

## Outputs
This is an internal error.

## Environment

- CovsirPhy version: 2.20.2
- Python version: 3.9.5
- Installation: poetry
- System: WSL (Ubuntu)

**Comment by lisphilar** [comment]
Datasets for cross validation are also shuffled by K-fold.

**Comment by lisphilar** [comment]
With #807, for CV, k-fold will be replaced with `TimeSeriesSplit`.
https://scikit-learn.org/stable/modules/cross_validation.html#time-series-split
https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html
